### PR TITLE
Adding changes to the governance and general issues voting

### PR DESCRIPTION
**Pull request by ocefpaf**
@scopatz and @CJ-Wright this is very crude but I'm sending this to get us started.

Basically my proposal is:

- both kind of voting must have public discussions;
- we can use the GH issues for discussing and voting, no need for secrecy on these two;
- the minimum time for voting should be at least one meeting cycle.

Does that make sense to you? Is it clear on the my test below?

### CODE REVIEW

File: src/governance.rst [hunk 1]
```diff
@@ -144,6 +144,24 @@ the time when a vote is called, but may never be less than 24 hrs.
...
+    for voting on the GH issue in question.
+    The voting must period be open for at least one core
+    member meeting cycle to allow for clarification questions
+    and discussions.
+
+    * Standard
+    * 50% Majority to pass
+
+:Modifying the governance document: The voting should happen in the PR
+    in question. The voting must period be open for at least one core
```
Comment (CJ-Wright) [inline src/governance.rst]: "The voting period must..."?

Comment (CJ-Wright) [reply src/governance.rst]: Maybe the PR also needs to have a `@conda-forge/core` call somewhere?

File: src/governance.rst [hunk 2]
```diff
@@ -144,6 +144,24 @@ the time when a vote is called, but may never be less than 24 hrs.
     * Sensitive
     * 75% Majority to pass
 
+:Overall workflow and packaging policies: The proposer can choose to
```
Comment (CJ-Wright) [inline src/governance.rst]: We might also require the policy to be added to the docs?

File: src/governance.rst (hunk 1 above)
Comment (CJ-Wright) [reply src/governance.rst]: Should we put this governance in its own repo so we can have review requirements?

Comment (ocefpaf) [reply src/governance.rst]: > Maybe the PR also needs to have a @conda-forge/core call somewhere?

File: src/governance.rst (hunk 2 above)
Comment (ocefpaf) [reply src/governance.rst]: This is rendered with the docs, right?

Comment (CJ-Wright) [reply src/governance.rst]: Sorry I meant that the new policy may go somewhere in the docs. It might be easier to read than gh issues.

File: src/governance.rst [hunk 3]
```diff
@@ -144,6 +144,25 @@ the time when a vote is called, but may never be less than 24 hrs.
...
+    * Standard
+    * 50% Majority to pass
+
+:Modifying the governance document: The voting should happen in the PR
+    in question and there mus tbe a call to `@conda-forge/core`.
+    The voting period must be open for at least one core
+    member meeting cycle to allow for clarification questions
+    and discussions.
+
+    * Sensitive
```
Comment (CJ-Wright) [inline src/governance.rst]: Sensitive votes are private, should this be standard?

File: src/governance.rst (hunk 2 above)
Comment (ocefpaf) [reply src/governance.rst]: I guess that depends on the policy and that will be part of the voting.
Like the `run_exports` one, we may want to write docs for it, but the "create a bot to close stale PRs on staged-recipes" we may not want to write docs.

File: src/governance.rst (hunk 3 above)
Comment (ocefpaf) [reply src/governance.rst]: You are right, maybe sensitive is not the best category here... Let's use Standard with 75%, is that OK?

File: src/governance.rst [hunk 4]
```diff
@@ -144,6 +144,25 @@ the time when a vote is called, but may never be less than 24 hrs.
...
 
+:Overall workflow and packaging policies: The proposer can choose to
+    create a poll with an external tool or call
+    for voting on the GH issue in question.
+    The voting must period be open for at least one core
+    member meeting cycle to allow for clarification questions
+    and discussions.
+
+    * Standard
+    * 50% Majority to pass
```
Comment (jakirkham) [inline src/governance.rst]: What about a threshold in terms of number of users out of the org participating?

Comment (msarahan) [reply src/governance.rst]: You can't force people to vote.  If anything, we should have rules about where polls should be posted, and how many reminders should be posted.  Beyond that, if people still don't participate, it should not invalidate a poll.

Comment (jakirkham) [reply src/governance.rst]: My concern is having enough buy-in for big changes. Otherwise they will be very hard to enact (even if a handful of people agree with them unanimously) or for that matter maintain.

**Comment by ocefpaf** [comment]
> but I am vehemently opposed to an absolute number of people required.

Indeed. We cannot force anyone to participate and vote. We can only give time and hope that those who are interest vote.

IMO one meeting cycle is not only enough time but also allow for some live discussion before voting.

Last but not least, nothing is writing in stone and any bad decision can be reverted if proven bad or harmful.

File: src/governance.rst (hunk 4 above)
Comment (jakirkham) [reply src/governance.rst]: If it's not going to attract enough attention to get some minimum number of votes, why does it need a vote at all? That suggests to me at least that the question isn't interesting and those involved can reasonably resolve it through a friendly discussion.

Comment (ocefpaf) [reply src/governance.rst]: Just take a look at the number of things we discuss in many meetings that do not attract attention and yet are blocked. Sorry, but the reality is not the scenario you described above. Also, why not vote after a discussion? Isn't that better than deciding on a closed meeting?

Comment (mwcraig) [reply src/governance.rst]: Based on my experience in our faculty senate and on several committees, I'd suggest a couple minor clarifications:

+ Let's clarify what "50%" (and 75% below) mean: I'd recommend changing the language to "50% of those voting to pass" and "75% of those voting to pass".
+ Maybe change "50%" to "50% plus one" so that when we have a tie vote we know what to do. If we don't decide this now then Murphy's law dictates we'll hit the situation in the first vote 😀 
+ @jakirkham is essentially suggesting we define what constitutes a quorum. That works very well for in-person meetings (and is arguably essential for good governance in that case); I understand the problem with compelling participation online, though. So how about this: `In the event that fewer than 50% of core members have participated in a vote by 24 hours before the deadline a reminder will be sent that they must vote to register approval or disapproval of the proposal. Failing to vote by the deadline will not affect the outcome.` This would set a goal of 50% participation, but not a block in the event we just don't get the responses. My guess in the event of very low participation is that most simply don't have an opinion (or don't understand the details of the proposal well enough to comment, at least in my case sometimes). I'm guessing those with strong opinions either way will vote.

Comment (ocefpaf) [reply src/governance.rst]: > + Let's clarify what "50%" (and 75% below) mean: I'd recommend changing the language to "50% of those voting to pass" and "75% of those voting to pass".

Good catch.

+ Maybe change "50%" to "50% plus one" so that when we have a tie vote we know what to do. If we don't decide this now then Murphy's law dictates we'll hit the situation in the first vote 😀 

:+1: 

> + @jakirkham is essentially suggesting we define what constitutes a quorum. That works very well for in-person meetings (and is arguably essential for good governance in that case); I understand the problem with compelling participation online, though. So how about this: `In the event that fewer than 50% of core members have participated in a vote by 24 hours before the deadline a reminder will be sent that they must vote to register approval or disapproval of the proposal. Failing to vote by the deadline will not affect the outcome.` This would set a goal of 50% participation, but not a block in the event we just don't get the responses. My guess in the event of very low participation is that most simply don't have an opinion (or don't understand the details of the proposal well enough to comment, at least in my case sometimes). I'm guessing those with strong opinions either way will vote.

The only change I see in your proposal is a burden of sending a reminder. I'm not against it but do we need more chores? We'll have 15 days, a meeting and the GH issues to interact on those during the voting period, do we really need a reminder 24 hours before the deadline?

Comment (scopatz) [reply src/governance.rst]: Maybe, "Friendly reminders are encouraged"

Comment (ocefpaf) [reply src/governance.rst]: > Maybe, "Friendly reminders are encouraged"

Sure. We kind of already do that already (at least I do :wink:)but we can write that in here.

Comment (mwcraig) [reply src/governance.rst]: Yeah, the burden of the reminder is a bit of a pain (and if it gets forgotten, then we'll get to boisterously discuss whether the vote counts if the reminder is forgotten).

What about clarifying (in all of the communications about these votes) that: voting yes moves the proposal forward; voting no is the only way to express opposition to the proposal; not voting  is discouraged, but  non-votes do not count as "no".

We could maybe add an explicit "Abstain" option to voting; I would choose that in some cases if it were available and it would give me or others a way to formally vote even if there are no strong feelings.

The problem with simply not voting is that "I never read or thought about the proposal or totally forgot to express an opinion" is indistinguishable from "I read it and don't have strong feelings either way".

Providing an "Abstain" option allows that, and it would be sensible to count those towards the "quorum".

Comment (ocefpaf) [reply src/governance.rst]: > We could maybe add an explicit "Abstain" option to voting; I would choose that in some cases if it were available and it would give me or others a way to formally vote even if there are no strong feelings.

Indeed that was missing and we must add that. Thanks!

**Comment by mwcraig** [comment]
@ocefpaf -- were you going to add to this doc that an "Abstain" option will be available in voting?

File: src/governance.rst [hunk 5]
```diff
@@ -144,6 +148,25 @@ the time when a vote is called, but may never be less than 24 hrs.
...
+    * 50% plus one of those voting to pass
+
+:Modifying the governance document: The voting should happen in the PR
+    in question and there must be a call to `@conda-forge/core`.
+    The voting period must be open for at least one core
+    member meeting cycle to allow for clarification questions
+    and discussions.
+
+    * Standard
+    * 75% of those voting to pass
```
Comment (mwcraig) [inline src/governance.rst]: Should this be "75% plus one" or "More than 75%" or "75% or more"? Don't have strong feelings about any of those options, but if the intent is that "3 out of 4 votes are yes ⇒ pass" then we should go with the last of those options.....

**Comment by jakirkham** [comment]
How do we decide whether a poll's text is ok? Is there a review process before sharing it?

File: src/governance.rst [hunk 6]
```diff
@@ -149,10 +153,10 @@ the time when a vote is called, but may never be less than 24 hrs.
     for voting on the GH issue in question.
     The voting must period be open for at least one core
     member meeting cycle to allow for clarification questions
-    and discussions.
+    and discussions. Friendly reminders to vote are encouraged.
 
     * Standard
-    * 50% Majority to pass
+    * 50% plus one of those voting to pass
```
Comment (dougalsutherland) [inline src/governance.rst]: Stupid clarity point: suppose 5 people vote. 50% is 2.5; 50% plus one is 3.5. I don't think that we want a 3/5 vote to fail. Maybe say instead `Strict majority` or `Over 50%` or something like that?

**Comment by ocefpaf** [comment]
> How do we decide whether a poll's text is ok? Is there a review process before sharing it?

The time for the discussions exists exactly for that.

File: src/governance.rst (hunk 6 above)
Comment (ocefpaf) [reply src/governance.rst]: Yeah, that is better.

**Comment by jakirkham** [comment]
Two points.

First people will inevitably miss meetings due to all manner of conflicts. It would be good to provide them an option to provide input despite inability to make a meeting.

Second discussing in the meeting and reviewing the text are not the same thing. It could be done in a shared doc if that is easier.

**Comment by isuruf** [comment]
Agree on @jakirkham's points. (Unrelated: there was an issue exactly like this on sagemath list. https://groups.google.com/forum/#!topic/sage-flame/JkQ0VwTUGfY)

**Comment by msarahan** [comment]
I agree that people not present in meetings need to have a voice, but that's hard to balance in discussions.  Meetings are where things are discussed in high-bandwidth environments, which is often essential for resolving contentious issues.  If a person makes their opinions in text and can't attend a meeting, how much weight does their opinion deserve?  Their opinion is by definition static in that meeting.  It can't block discussion in the meeting outright.  When it is strongly opposed to the general consensus reached in the meeting, how can that be reconciled?  I think that is why decisions should be made by vote, not necessarily in meetings - it allows for asynchronicity.  Still, there is a cost to not attending meetings, and I'm not sure there's any way around that.  We can't necessarily completely make up for a very high-bandwidth discussion with lots and lots of low-bandwidth discussions.

**Comment by ChrisBarker-NOAA** [comment]
+    * 50% plus one of those voting to pass

Couldn’t it simply be >= X% ?

-CHB

**Comment by mwcraig** [comment]
How about circulating the text for review (with a deadline)? We could have a repo for poll questions (and results if we want) with new ones opened as PRs. 

I don’t think this replaces discussion at a meeting, but post-draft, pre-sending-out of polls/surveys almost always catches typos/lack of clarity even after extensive discussion. 

Agreed this needs to have a short turnaround, and will be best suited to small changes, not a re-discussion of the broader poll.

**Comment by ocefpaf** [comment]
> Agreed this needs to have a short turnaround, and will be best suited to small changes, not a re-discussion of the broader poll.

Even with a short turnaround I'm afraid this extra step is just another hill for ideas to die at. The goal of this PR is to make conda-forge more agile and avoid the "bureaucratic halts" we are facing lately.

**Comment by mwcraig** [comment]
> Even with a short turnaround I'm afraid this extra step is just another hill for ideas to die at. The goal of this PR is to make conda-forge more agile and avoid the "bureaucratic halts" we are facing lately.

Understood, but what about posting the text right after the meeting with a 48 hour deadline (if necessary no vote == approval), or posting for 48 hours prior to the meeting?

**Comment by jakirkham** [comment]
> Unrelated: there was an issue exactly like this on sagemath list. https://groups.google.com/forum/#!topic/sage-flame/JkQ0VwTUGfY

This has also come up in regards to [Python's PEP process]( https://mail.python.org/pipermail/python-committers/2018-July/005713.html ). There's a bunch of other points that come up in that thread. It's worth reading if for no other reason than to understand how Python's governance model is shifting and how we might learn from it.

**Comment by scopatz** [comment]
Hi All, first off, I want to thank everyone for the time and effort and thought that they have put into this. I think that their are a few things that we should acknowldege in considering this.

1. We currently have no offical way to modify the governance docuement or to make policy decisions (my bad!).
2. I completely agree with @embray that the way questions are phrased can be purposefully (mis)leading. 
3. Raymod Hettinger's idea about having documents that capture both the pro & con thought processes seems to be very valueable, as it allows us to diagree on an issue in a civil, and structured manner, while actually making decisions. 
4. We all want conda-forge to succeed and be a positive and attractive space for people to contribute (which is why we have a code-of-conduct, give out mugs, etc)

I think that the remaining questions for this PR are issues that are solvable by merging this PR.  For example, @mwcraig's question of, 

> what about posting the text right after the meeting with a 48 hour deadline (if necessary no vote == approval), or posting for 48 hours prior to the meeting?

is something that we could have a vote to amend the governance document on if this PR were merged. Similarly, adopting ideas from the Python PEP process like Hettinger's are clear, explcit items we can vote on if this PR is merged. 

I actually think the concerns about leading questions are already addressed in the governance document itself.  We already have in the document, prior to this PR, the statement that `The act of calling for a vote cannot itself violate the code of conduct.`  If someone - even a member of core - is attempting to abuse these privldeges, we have several mechanisms for recourse. 

That said, I don't beleive that anyone currently on core would abuse their status or bully people in order to get what they want. I don't think this is a problem we have or are likely to have. Our technical disagreements have been just that, and have not been personal or malicious.

So in summary, I believe that

* These changes let us move forward with further ammendments to the document and other key decision making, and 
* While we need to be cognizant of potential abuses of power, we also have little reason to suspect anyone of abusing their power at this point.
* These are close enough the minimal changes needed to move forward.
* We should not try to stick too many changes into a single PR, especially since we don't yet have a defined process.

If there are no major show-stopper objections to the text as it stands, and with the above points in mind, I plan on merging this PR tomorrow in order to give us all a more structured way to proceed.

Thanks again to everyone for all of the care and attention you have given to this issue.

File: src/governance.rst [hunk 7]
```diff
@@ -144,6 +148,25 @@ the time when a vote is called, but may never be less than 24 hrs.
     * Sensitive
     * 75% Majority to pass
 
+:Overall workflow and packaging policies: The proposer can choose to
+    create a poll with an external tool or call
+    for voting on the GH issue in question.
+    The voting must period be open for at least one core
```
Comment (jakirkham) [inline src/governance.rst]: `The voting must period be open...` -> `The voting period must be open...`

**Comment by jakirkham** [comment]
Clarifying point: My understanding thus far is core is the only group that gets to vote on things as there is no specification for votes from anyone else. Is that correct?

**Comment by dougalsutherland** [comment]
> I actually think the concerns about leading questions are already addressed in the governance document itself. [...] That said, I don't beleive that anyone currently on core would abuse their status or bully people in order to get what they want.

I think there's a difference between "I'm bullying people into making my decision" and "I put out a poll without carefully considering the wording that ended up biased towards my point of view." The first one we have some recourse for and feels not particularly likely, but the second one is very easy to see happening, and would then call into question the whole decision / etc....

File: src/governance.rst (hunk 7 above)
Comment (scopatz) [reply src/governance.rst]: Thanks!

**Comment by scopatz** [comment]
> Clarifying point: My understanding thus far is core is the only group that gets to vote on things as there is no specification for votes from anyone else. Is that correct?

@jakirkham - yes, this is mentioned earlier in the document in the "Voting" section

> The core team is the only team with voting rights.

------ 

> I think there's a difference between "I'm bullying people into making my decision" and "I put out a poll without carefully considering the wording that ended up biased towards my point of view." The first one we have some recourse for and feels not particularly likely, but the second one is very easy to see happening, and would then call into question the whole decision / etc....

@dougalsutherland - yeah I agree that those are different cases. But it does seem to me that  the proposals that the proposals we seen come through so far have had a lot of thought put into the wording and discussion about the wording. For example, the compiler migration proposal had its own document that was a few pages long and listed all of the options. While others might not have been as verbose, they certainly had a lot of discussion before and after the vote was called for.

I think if it gets to the point where we are having egregiously thoughtless proposals come in, we should vote on changes to the governance document to prevent such polls coming through in the future. But again, as things stand now, we have no mechanism for ammending the governance document. So merging this issue will allow us to move forward with these and any other concerns we may have.

**Comment by scopatz** [comment]
Thanks again everyone!

**Comment by embray** [comment]
I was confused as to why @scopatz was agreeing with me since I didn't recall taking part in this discussion earlier.  But I'm glad--reading the discussion on this PR, as well as the PR itself was interesting--as is conda-forge's excellent (nitpicks aside) project governance documentation.  I hope one day we can have something similar for Sage...
